### PR TITLE
testdisk 7.0 and basic test added

### PR DESCRIPTION
**Pull request by alistairmcmillan**
testdisk updated to 7.0.
Added basic test.
Syntax updates prompted by audit --strict.

### CODE REVIEW

File: Library/Formula/testdisk.rb [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
                           "--prefix=#{prefix}"
-    system "make install"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/testdisk", "/help"
```
Comment (mikemcquaid) [inline Library/Formula/testdisk.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

Comment (alistairmcmillan) [inline Library/Formula/testdisk.rb]: I knew you'd say that. :)

Not sure what I can do for a test here except create a little disk image with hdiutil and then get test disk to "/list" the structure. Would that be okay?

Comment (mikemcquaid) [inline Library/Formula/testdisk.rb]: Yup :+1:. Anything with `/list` sounds good!

**Comment by alistairmcmillan** [comment]
I'm not sure what the error means on Mountain Lion. hdiutil did exist on Mountain Lion  and the parameters don't seem to have changed. Can anyone point me in the right direction?

File: Library/Formula/testdisk.rb [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
+    system "make", "install"
+  end
+
+  test do
+    path = testpath/"testdisk.dmg"
+    args = %W[
+      create
+      -megabytes 10
+    ]
+    system "hdiutil", *args, path
```
Comment (mikemcquaid) [inline Library/Formula/testdisk.rb]: `*args` needs to be the last argument with Ruby 1.8 (i.e. Mountain Lion and below)

Comment (alistairmcmillan) [inline Library/Formula/testdisk.rb]: Thanks.

File: Library/Formula/testdisk.rb [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
+      create
+      -megabytes 10
+      testpath/"testdisk.dmg"
+    ]
+    system "hdiutil", *args
+    args = %W[
+      /list
+      testpath/"testdisk.dmg"
+    ]
+    system "#{bin}/testdisk", *args
```
Comment (mikemcquaid) [inline Library/Formula/testdisk.rb]: I think it's probably nicer to just inline these args lists. If you want to use a variable for something I think it's worth doing for the dmg name. Also, you probably don't need to specify `testpath` as that's the current directory.

**Comment by alistairmcmillan** [comment]
Fifth times a charm. Thanks for your help Mike. Let me know if there is anything else I need to change before this can be accepted.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by mikemcquaid** [comment]
For future reference the preferred commit message format for simple version updates is `testdisk 7.0`.

**Pull request closed without merging**